distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso

```diff
@@ -1362,6 +1370,306 @@ Text.last_index_of self term="" start=-1 case_sensitivity=Case_Sensitivity.Sensi
+
+   Parses a textual representation of an integer into an integer number, returning
+   a `Number_Parse_Error` if the text does not represent a valid integer.
+
+   Arguments:
+   - radix: The number base to use for parsing (defaults to 10).
+
+   > Example
+     Parse the text "20220216" into an integer number.
+
+        "20220216".parse
+Text.parse_integer : Integer -> Integer ! Number_Parse_Error
+Text.parse_integer self (radix=10) = Integer.parse_builtin self radix
+
+## ALIAS JSON From Text
+
+   Parse a Text value into a `JS_Object` or an Enso primitive value (like
+   `Text`, `Number`, `Boolean`, `Nothing`), or a `Vector` of values.
+
+   > Example
+    Parse the text "[null, null, true, false]".
+
+        "[null, null, true, false]".parse_json
+Text.parse_json : JS_Object | Boolean | Number | Nothing | Text | Vector ! Invalid_JSON
+Text.parse_json self = Json.parse self
+
+## ALIAS Date from Text
+
+   Converts text containing a date into a Date object.
+
+   Arguments:
+   - format: An optional format describing how to parse the text.
+
+   Returns a `Time_Error` if `self`` cannot be parsed using the provided
+   `format`.
+
+   ? Format Syntax
+     Formats are based on a simple sequence of letters and symbols. For
+     example, "d MMM yyyy" will format "2011-12-03" as "3 Dec 2011".
+
+   ? Default Date Formatting
+     Unless you provide a custom format, the text must represent a valid date
+     that can be parsed using the ISO-8601 extended local date format. The
+     format consists of:
+
+     - Four digits or more for the year. Years in the range 0000 to 9999
+       will be pre-padded by zero to ensure four digits. Years outside
+       that range will have a prefixed positive or negative symbol.
+     - A dash
+     - Two digits for the month-of-year. This is pre-padded by zero to ensure
+       two digits.
+     - A dash
+     - Two digits for the day-of-month. This is pre-padded by zero to ensure two
+       digits.
+
+   > Example
+     Parse the date of 23rd December 2020.
+
+         import Standard.Base.Data.Text.Extensions
+
+         example_parse = "2020-12-23".parse_date
+
+   > Example
+     Recover from an error due to a wrong format.
+
+         import Standard.Base.Data.Text.Extensions
+         from Standard.Base.Errors.Common import Time_Error
+
+         example_parse_err = "my birthday".parse_date . catch Time_Error _->
+             Date.new 2000 1 1
+
+   > Example
+     Parse "1999-1-1" as Date using a custom format.
+
+         import Standard.Base.Data.Text.Extensions
+
+         example_parse = "1999-1-1".parse_date "yyyy-M-d"
+
+   > Example
+     Recover from the parse error.
+
+         import Standard.Base.Data.Text.Extensions
+         from Standard.Base.Errors.Common import Time_Error
+
+         example_parse_err =
+             date = "1999-1-1".parse_date "yyyy-MM-dd"
+             date.catch Time_Error (_->Date.new 2000 1 1)
+Text.parse_date : (Text | Nothing) -> Date ! Time_Error
```

Comment:
While not really this task we should add `Locale` support to `parse_date`

Reply:
I'll add locale to Date here: https://github.com/enso-org/enso/issues/6424